core/src/main/java/co/aikar/commands/CommandIssuer.java

```diff
@@ -50,6 +53,11 @@ default void sendMessage(String message) {
         getManager().sendMessage(this, MessageType.INFO, MessageKeys.INFO_MESSAGE, "{message}", message);
     }
 
+    /**
+     * @return the UUID of that player or Optional.empty() if it's not a player
+     */
+    Optional<UUID> getUniqueId();
```

Comment:
We don't use Optionals in ACF. I'm actually going to say lets make this NotNull also.

For JDA, we can use UUID.fromString(String.valueOf(user.userId));

This way we can get a consistent way to identify a user, to be used by other future things I have planned.

Even if an implementation doesnt have authentication to its commands, they can either use session ID's, ie a UUID per connection to the server (RCON),  or better yet, per ip address.

And worst case, random. But it's easy to get creative on identifying same user.

Reply:
additionally, add in some validation on creation of Command Issuers (ie where the .createCommandIssuer method is called), that checks .getUniqueId() for null and throws an error "Command Issuer did not return a UUID".

And this method can default MethodNotImplemented exception

Reply:
>For JDA, we can use UUID.fromString(String.valueOf(user.userId));

I guess you mean `UUID.nameUUIDFromBytes(user.getId().getBytes(StandardCharsets.UTF_8))`, because `UUID.fromString` requires a 128 bit UUID string like this `0aaa2c13-922a-411b-b655-9b8c08404695`. 

I implemented the first one now for non-players in game platforms (Sponge, Bukkit, BungeeCord) using the sender name (i.e "Rcon" or "Console"). 

For JDA we could use a constant `0L` prefix and use `long user.getIdLong()` for the least significant bits. (`new UUID(0L, userId`)